drizzlepac/hlautils/config_utils.py

```diff
@@ -190,15 +189,8 @@ def _determine_conditions(self, prod_obj):
 
     def _get_cfg_index(self):
         """return the contents of the appropriate index cfg file."""
-        code_dir = os.path.abspath(__file__)
-        base_dir = os.path.dirname(os.path.dirname(code_dir))
-        self.pars_dir = os.path.join(base_dir, "pars/hap_pars")
-        cfg_index_fileanme = self.inst_det + "_index.json"
-        cfg_index_filename = os.path.join(self.pars_dir, cfg_index_fileanme)
-
-        with open(cfg_index_filename) as jsonFile:
-            json_string = jsonFile.read()
-            self.full_cfg_index = json.loads(json_string)
+        self.full_cfg_index, self.pars_dir = read_index(self.instrument, 
+                                                        self.detector)
```

Comment:
Class attributes should at the least be declared in the init of the class and initialized to a default value (e.g., None). In this way the reader knows there is a class attribute which has yet to be computed.  When feasible, attributes computed by methods of a class should return the values rather than setting the values in the method in order to avoid "hiding" the setting of the values.